### PR TITLE
Create glossary.md for the documentation sprint #1040

### PR DESCRIPTION
**Pull request by manez**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/1040

# What does this Pull Request do?

Adds a new page with a starter glossary of terms for Islandora 8. Based off the glossaries of Islandora 7, Drupal 8, and a scan of our documentation.

# How should this be tested?

Preview the page and look it over. All definitions ok? Typos and formatting ok?

# Additional Notes:
Happy to accept suggestions for terms that should be added!

There's no reviewer [signed up yet](https://docs.google.com/spreadsheets/d/1E-kRw9xE60CKK0qL1-phzeVKjEZu3qBKZ9d3LH1hDEE/), so whoever reviews, please grab the credit in that spreadsheet.

# Interested parties
@Islandora/8-x-committers

### CODE REVIEW

File: docs/user-documentation/glossary.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+The following glossary of terms addresses an Islandora 8 context. When comparing new Islandora and Fedora to older versions it may also be helpful to reference [the Islandora 7 Glossary](https://wiki.duraspace.org/display/ISLANDORA/APPENDIX+E+-+Glossary).
+ 
+## CLAW
+ CLAW (CLAW Linked Asset WebDataFrameWork) was the production code name for the software released in June, 2019 as _Islandora 8_
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: Wouldn't it be 'development code name'? Also, we need a period at the end.

File: docs/user-documentation/glossary.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+## Drupal 
+Drupal is an open source content management system (CMS) written in PHP. Known for being extremely flexible and extensible, Drupal is supported by a community of over 630,000 users and developers. Drupal sites can be customized and themed in a wide variety of ways.
+ 
+Drupal functions as the _front-end_ of Islandora.
+ 
+## Drupal Core
+The files, themes, profiles, and modules included with the standard project software download.
+ 
+## Drupal Roles
+Roles are a way of assigning specific permissions to a group of users. Any user assigned to a role will have the same permissions as all other users assigned to that role. This allows you to control which users have permission to view, edit, or delete content in both Drupal and the Fedora repository.
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: > This allows you to control which users have permission to view, edit, or delete content in both Drupal and the Fedora repository.

The Roles allow you to control what actions members of that role can perform in Drupal; however, it is membership in the fedoraAdmin role that causes those actions to be reflected in Fedora. (E.g. you can add me to a role that allows me to create resource nodes; but unless I'm a member of fedoraAdmin that new node won't be indexed.) The fedoraAdmin role doesn't technically need any permissions to serve its purpose.

File: docs/user-documentation/glossary.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Fedora (Repository Software)
+Fedora is a digital asset management architecture upon which institutional repositories, digital archives, and digital library systems might be built. Fedora is the underlying architecture for a digital repository, and is not a complete management, indexing, discovery, and delivery application. 
+ 
+The Fedora repository functions as the standard _smart storage_ for Islandora.
+ 
+## Field
+Data of a certain type that is attached to a content entity. For instance, on a Resource Node content type, you might have fields for a title, description, display hints, subjects, and other metadata. 
+ 
+## Flysystem
+Flysystem is a filesystem abstraction library for PHP. Islandora uses Flystsem to swap about different backend filesystem applications, and comes with a custom Flystsem adapter for Fedora.
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: "filesystem applications, and comes with a custom Flystsem adapter for Fedora." should probably be "filesystem applications. Islandora provides a custom Flystsem adapter for Fedora."

File: docs/user-documentation/glossary.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+A drop-in replacement for the Drupal core upload field with a unified User Interface where editors and administrators can upload, manage, and reuse files and multimedia assets. 
+ 
+## Microservice
+A software development technique —a variant of the service-oriented architecture (SOA) structural style— that arranges an application as a collection of loosely coupled services[1] In a microservices architecture, services are fine-grained and the protocols are lightweight.
+ 
+## Open Source
+Open source describes a method of software development that promotes access to the end product's source code. Islandora is an open source product with an active development community, operating under the GPL license (2.0) for Drupal components and the MIT license for non-Drupal components. 
+ 
+## Module
+Software (usually PHP, JavaScript, and/or CSS) that extends site features and adds functionality. The Drupal project distinguishes between core and contributed modules.
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: "Drupal modules conform to a specific structure allowing them to integrate with the Drupal architecture."

File: docs/user-documentation/glossary.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ 
+## Node
+A node is any piece of individual content, such as a page, poll, article, forum topic, or a blog entry. All content on a Drupal website is stored and treated as "nodes". For information about nodes specific to Islandora, see "Resource Node."
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: > Unlike Article and Basic page, Poll is not a content type that creates nodes with fields like it did in the past. "Poll" is an entity, it is the question itself.

See https://www.drupal.org/docs/8/modules/poll

File: docs/user-documentation/glossary.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+ 
+## Resource Node
+A resource node is a Drupal node that represents a single conceptual item or object stored in an Islandora repository. It acts as a container for all files and metadata associated with that item, and is the place where the item 'lives' as a visitable url. 
+ 
+For example, A video stored in Islandora will have a resource node, which metadata stored in fields, a preservation master file, and may have a thumbnail, web-friendly derivatives, and technical metadata associated with the resource node. The resource node may also belong to one or more collections. 
+ 
+## Role
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: Duplicated above as "Drupal Role"

File: docs/user-documentation/glossary.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+A named set of permissions that can be applied to a user account. 
+ 
+## Taxonomy term
+A term used to classify content, such as a tag or a category. 
+ 
+## Theme
+Software and asset files (images, CSS, PHP code, and/or templates) that determine the style and layout of the site. The Drupal project distinguishes between core and contributed themes. 
+ 
+## View 
+A Drupal term for a formatted listing of data; typically, the data comes from content entities.
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: "An abstract query configuration used to generate lists and tables of content entities or specific content entity fields. Drupal provides a powerful administrator interface for creating and editing View configurations."

File: docs/user-documentation/glossary.md (hunk 1 above)
Comment (manez) [reply docs/user-documentation/glossary.md]: 🤦‍♀ yes, _development_

File: docs/user-documentation/glossary.md (hunk 5 above)
Comment (dannylamb) [reply docs/user-documentation/glossary.md]: Learn something new every day....

File: docs/user-documentation/glossary.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+For example, A video stored in Islandora will have a resource node, which metadata stored in fields, a preservation master file, and may have a thumbnail, web-friendly derivatives, and technical metadata associated with the resource node. The resource node may also belong to one or more collections. 
+ 
+## Taxonomy term
+A term used to classify content, such as a tag or a category. 
+ 
+## Theme
+Software and asset files (images, CSS, PHP code, and/or templates) that determine the style and layout of the site. The Drupal project distinguishes between core and contributed themes. 
+ 
+## View 
+An abstract query configuration used to generate lists and tables of content entities or specific content entity fields. Drupal provides a powerful administrator interface for creating and editing View configurations.
```
Comment (dannylamb) [inline docs/user-documentation/glossary.md]: I'd slim this down to

> A database query used to generate lists or tables of content.  Drupal provides a powerful administrator interface for creating and editing views without any coding.

File: docs/user-documentation/glossary.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+## Fedora (Repository Software)
+Fedora is a digital asset management architecture upon which institutional repositories, digital archives, and digital library systems might be built. Fedora is the underlying architecture for a digital repository, and is not a complete management, indexing, discovery, and delivery application. 
+ 
+The Fedora repository functions as the standard _smart storage_ for Islandora.
+ 
+## Field
+Data of a certain type that is attached to a content entity. For instance, on a Resource Node content type, you might have fields for a title, description, display hints, subjects, and other metadata. 
+ 
+## Flysystem
+Flysystem is a filesystem abstraction library for PHP. Islandora uses Flystsem to swap about different backend filesystem applications. Islandora provides a custom Flystsem adapter for Fedora.
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/glossary.md]: Whoops, typo: two instances of "Flystsem" -> "Flysystem".

**Comment by seth-shaw-unlv** [comment]
Other than _poll_ being listed as a type of node, the Flysystem typos, and @dannylamb' edit; this looks good.

However, I would feel better about merging if we got one of the newer folks to take a look and make sure the definitions are clear to them.

**Comment by manez** [comment]
I've made those updates. I'll send a note out to the list of sprinters to see if any of the others might be willing to give this a look over. New eyes is a good idea.

**Review by seth-shaw-unlv — APPROVED** [review]
Looks good to me.

**Comment by cderven** [comment]
As a newer Islandora 8 person/ person on the periphery, the definitions all make sense to me, Melissa. Formatting, etc. look OK.

Under "Microservices" there may be a typo. There's currently an endnote/ footnote "[1]" after "loosely coupled services". Should this be a full stop?

**Comment by manez** [comment]
@cderven quite right, that reference went nowhere. Fixed now!

@seth-shaw-unlv I this this might be ready to merge.

**Pull request merged**